torch/ao/quantization/fx/utils.py

```diff
@@ -455,6 +456,78 @@ def all_node_args_have_no_tensors(node: Node, modules: Dict[str, torch.nn.Module
         cache[node] = result
     return result
 
+def all_node_args_except_first(node: Node) -> List[int]:
+    """
+    Returns all node arg indices after first
+    """
+    return [i for i, arg in enumerate(node.args)][1:]
+
+def return_arg_list(arg_indices: List[int]) -> Callable[[Node], List[int]]:
+    """
+    Constructs a function that takes a node as arg and returns the arg_indices
+    that are valid for node.args
+    """
+    def arg_indices_func(node: Node) -> List[int]:
+        result = []
+        for i in arg_indices:
+            if i < len(node.args):
+                result.append(i)
+        return result
+    return arg_indices_func
+
+NodeInfo = namedtuple("NodeInfo", "op target")
+
+# this dict identifies which indices of a node are non tensors
+# so that they can be propagated correctly since inserting observers
+# for them would cause errors
+
+NON_OBSERVABLE_ARG_DICT: Dict[NodeInfo, Dict[Union[type, torch.dtype], Callable[[Node], List[int]]]] = {
```

Comment:
wondering why don't we have a map from index to arg type? I feel that is probably a bit more intuitive?

Reply:
I considered something like that but I wasn't sure whether there was a more efficient way to do that than inserting each index to a dict one at a time. This way I create a dict, the key is the dtype and the I insert a single list which seemed more efficient for both creation and searching.

Reply:
I see, if we want to support all the ops we may need to get this information in somewhere instead of handwritten them I think, I'll ask around later about this info to support this generally.